hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/ReplicatedMapSplitBrainHandlerService.java

```diff
@@ -90,17 +106,136 @@ public Runnable prepareMergeRunnable() {
 
         private static final int TIMEOUT_FACTOR = 500;
 
-        Map<String, Collection<ReplicatedRecord>> recordMap;
+        private final Semaphore semaphore = new Semaphore(0);
+        private final ILogger logger = nodeEngine.getLogger(ReplicatedMapSplitBrainHandlerService.class);
+        private final Map<String, Collection<ReplicatedRecord>> recordMap;
 
         Merger(Map<String, Collection<ReplicatedRecord>> recordMap) {
             this.recordMap = recordMap;
         }
 
         @Override
         public void run() {
-            final ILogger logger = nodeEngine.getLogger(ReplicatedMapService.class);
-            final Semaphore semaphore = new Semaphore(0);
+            int recordCount = 0;
+            for (Map.Entry<String, Collection<ReplicatedRecord>> entry : recordMap.entrySet()) {
+                String name = entry.getKey();
+                ReplicatedMapConfig replicatedMapConfig = service.getReplicatedMapConfig(name);
+                Object mergePolicy = getMergePolicy(replicatedMapConfig);
+                if (mergePolicy instanceof SplitBrainMergePolicy) {
+                    int batchSize = MergePolicyConfig.DEFAULT_BATCH_SIZE;
```

Comment:
Is a cluster version check missing here? Only 3.10+ can handle the unified split brain merge policies.

Reply:
agree

Reply:
Whoops, totally missed that. Also the `batchSize` should be taken from the `MergePolicyConfig`. Good catch!